--- a/averagingwriter.cpp
+++ b/averagingwriter.cpp
@@ -1,8 +1,10 @@
 #include "averagingwriter.h"
 
+#if defined(__i386__) || defined(__x86_64__) || defined(_M_IX86) || defined(_M_X64)
 #include <xmmintrin.h>
 
 #define USE_SSE
+#endif
 
 void AveragingWriter::WriteRow(double time, double timeCentroid, size_t antenna1, size_t antenna2, double u, double v, double w, double interval, const std::complex<float>* data, const bool* flags, const float *weights)
 {
--- a/cotter.cpp
+++ b/cotter.cpp
@@ -24,9 +24,11 @@
 #include <cmath>
 #include <complex>
 
+#if defined(__i386__) || defined(__x86_64__) || defined(_M_IX86) || defined(_M_X64)
 #include <xmmintrin.h>
 
 #define USE_SSE
+#endif
 
 using namespace aoflagger;
 
